src/plugins/part_err.c

```diff
@@ -17,15 +17,24 @@
  * Author: Vojtech Trefny <vtrefny@redhat.com>
  */
 
+#include <glib.h>
 #include <parted/parted.h>
 
 #include "part_err.h"
 
 static __thread gchar *error_msg = NULL;
 
 PedExceptionOption bd_exc_handler (PedException *ex) {
-    error_msg = g_strdup (ex->message);
-    return PED_EXCEPTION_UNHANDLED;
+    /*
+     * Only treat exceptions as errors if they aren't at least errors or can't
+     * be ignored.
+     */
+    if (ex->type >= PED_EXCEPTION_ERROR || (ex->options & PED_EXCEPTION_IGNORE) == 0) {
+      error_msg = g_strdup (ex->message);
+      return PED_EXCEPTION_UNHANDLED;
+    }
+    g_warning ("[parted] %s", ex->message);
+    return PED_EXCEPTION_IGNORE;
```

Comment:
I'd prefer if the logic was inverted here. So something like ``if THIS_CAN_BE_IGNORED -> ignore; else raise``